tools/c7n_gcp/c7n_gcp/actions/iampolicy.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n.utils import type_schema
+from c7n_gcp.actions import MethodAction
+
+
+class SetIamPolicyBaseAction(MethodAction):
+
+    schema = type_schema('set-iam-policy',
+                         required=['bindings'],
+                         **{
+                             'bindings': {
+                                 'type': 'array',
+                                 'minItems': 1,
+                                 'items': {'role': {'type': 'string'},
+                                           'members': {'type': 'array',
+                                                       'items': {
+                                                           'type': 'string'},
+                                                       'minItems': 1}}
+                             }
+                         })
+    method_spec = {'op': 'setIamPolicy'}
```

Comment:
set `schema_alias` to true so the schema for this is only registered once even across multiple resources (the other resources get jsonpointer aliases to it in the schema then). it also properly shows up in the docs as a common action (for which a doc string here would be good).

Reply:
Thank you. Fixed